normandy/recipes/models.py

```diff
@@ -268,34 +271,24 @@ def data(self):
 
     @property
     def filter_expression(self):
-        exp = ''
+        parts = []
 
         if self.locales.count():
             locales = ', '.join(["'{}'".format(l.code) for l in self.locales.all()])
-            exp = 'normandy.locale in [{}]'.format(locales)
+            parts.append('normandy.locale in [{}]'.format(locales))
 
         if self.countries.count():
             countries = ', '.join(["'{}'".format(c.code) for c in self.countries.all()])
-
-            if len(exp):
-                exp += ' && '
-
-            exp += 'normandy.country in [{}]'.format(countries)
+            parts.append('normandy.country in [{}]'.format(countries))
 
         if self.channels.count():
             channels = ', '.join(["'{}'".format(c.slug) for c in self.channels.all()])
-
-            if len(exp):
-                exp += ' && '
-
-            exp += 'normandy.channel in [{}]'.format(channels)
+            parts.append('normandy.channel in [{}]'.format(channels))
 
         if self.extra_filter_expression:
-            prefix = ' && (' if len(exp) else ''
-            suffix = ')' if len(exp) else ''
-            exp += '{}{}{}'.format(prefix, self.extra_filter_expression, suffix)
+            parts.append(self.extra_filter_expression)
 
-        return exp
+        return ' && '.join(parts)
```

Comment:
I think you still want to surround the parts with parentheses just to be safe, just do it for all of them always.